### DISCUSSION TITLE
Can GPU failures be handled gracefully?

### DISCUSSION THREAD

**Discussion by dsjstc** [blakeblackshear/frigate#8589] [Ask A Question]
I run Frigate in a on my basement server, which also gets used for a lot of things, including ML development.  Sometimes the GPU is being used heavily for other things.

I'd love to be able to (temporarily) turn off the GPU-dependent features (or simply let them stop working if the GPU is unavailable or out of memory).  Basically, it'd be nice if Frigate could keep recording in spite of a GPU failure.

Is there any way for me to set that up?

**Comment by NickM-27** [comment]
Detection is dependent on decoding so there's no easy way for that. 

Recording doesn't use the GPU so just need to make sure it doesn't have hwaccel args set for that input. If you show your config I can make a suggestion

**Reply by dsjstc** [reply]
Thanks for that, I thought that all of frigate was shutting down due to detection failing on GPU out-of-memory.  I don't think that anymore.

Basically I think the question boils down to "is there a way for ffmpeg to use hwaccel on an as-available basis?"  After more review of the docs, I don't see anything like that.  So for now I'll just disable it.

Thanks for your comment, I appreciate it!
